### PR TITLE
fix(ecma/parser): throw error when function body has use strict and paramaters is not simple

### LINKED ISSUE #3236: Use Strict directive in function with non-simple paramaters should throw error
### Describe the bug

In accordance with ES standards, the "use strict" directive is not allowed to be used within functions with non-simple paramaters. The [input script](https://github.com/tc39/test262/blob/6e61dd7754e7c94ebcf3ee679cb8db3c54a37b50/test/language/statements/function/use-strict-with-non-simple-param.js) was taken from this test262 script.

### Input code

```typescript
function f(a = 0) {
  "use strict";
}
```

### Config

_No response_

### Playground link

https://play.swc.rs/?version=1.2.124&code=H4sIAAAAAAAAA0srzUsuyczPU0jTSFSwVTDQVKjmUlBQKi1OVSguKcpMLlGy5qoFAHt1YaIlAAAA&config=H4sIAAAAAAAAA0WMSwrDMAxE76K1t93kDj2EcJXg4B8aBWqM714bUrKSmPdmOp3wtHWqrBBdH1o2%2FtJG4hPDa6hGbmoz2jlChiNjPcSWgtdksRTITR2lkMPe1pIvqaoAD%2BJ8xL855lAqn2sFnazVeVcllXyCxjN0lwPet216yfgBCKaI9boAAAA%3D

### Expected behavior

An error similar to this one from babel should be thrown

Illegal 'use strict' directive in function with non-simple parameter list. (1:0)

> 1 | function f(a = 0) {
    | ^
  2 |   "use strict";
  3 | }

### Actual behavior

The script transpiles with no exceptions and allows the "use strict" directive within the function.

### Version

1.2.127

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by RiESAEX**
**Description:**
throw SyntaxError: Illegal 'use strict' directive in function with non-simple parameter list,
when function body has a 'use strict' and function paramaters is not simple

**Related issue (if exists):**
fix #3236

### CODE REVIEW

File: crates/swc_ecma_parser/src/parser/class_and_fn.rs [hunk 1]
```diff
@@ -1373,26 +1382,77 @@ impl OutputType for Decl {
...
+        self.iter().all(|param| {
+            matches!(
+                param,
+                ParamOrTsParamProp::TsParamProp(..)
+                    | ParamOrTsParamProp::Param(Param {
+                        pat: Pat::Ident(_),
+                        ..
+                    })
+            )
+        })
```
Comment (RiESAEX) [inline crates/swc_ecma_parser/src/parser/class_and_fn.rs:1455]: not sure about ts params, so I ignored them

**Review by kdy1 — CHANGES_REQUESTED** [review]
Can you add some span tests?
For now, you can add typescript files at https://github.com/swc-project/swc/tree/d396c32fe5139fdfa04934e3f29297ac1a8e99f7/crates/swc_ecma_parser/tests

**Comment by RiESAEX** [comment]
I don't understand, what is span test?
btw I just found out that tsc doesn't emit this error

**Comment by kdy1** [comment]
I meant something like https://github.com/swc-project/swc/blob/d396c32fe5139fdfa04934e3f29297ac1a8e99f7/crates/swc_ecma_parser/tests/typescript-errors/issue-1209/input.ts.stderr

**Review by kdy1 — APPROVED** [review]

**Pull request merged**